transaction/core/src/ring_signature/mlsag.rs

```diff
@@ -332,19 +356,50 @@ fn challenge(
 }
 
 fn decompress_ring(
-    ring: &[(CompressedRistrettoPublic, CompressedCommitment)],
-) -> Result<Vec<(RistrettoPublic, Commitment)>, Error> {
+    ring: &[ReducedTxOut],
+) -> Result<Vec<(RistrettoPublic, Commitment)>, MLSAGError> {
     // Ring must decompress.
     let mut decompressed_ring: Vec<(RistrettoPublic, Commitment)> = Vec::new();
-    for (compressed_address, compressed_commitment) in ring {
-        let ristretto_public =
-            RistrettoPublic::try_from(compressed_address).map_err(|_e| Error::InvalidCurvePoint)?;
-        let commitment = Commitment::try_from(compressed_commitment)?;
+    for tx_out in ring {
+        let ristretto_public = RistrettoPublic::try_from(&tx_out.target_key)
+            .map_err(|_e| MLSAGError::InvalidCurvePoint)?;
+        let commitment = Commitment::try_from(&tx_out.commitment)?;
         decompressed_ring.push((ristretto_public, commitment));
     }
     Ok(decompressed_ring)
 }
 
+/// An error which can occur when signing or verifying
+#[derive(Clone, Debug, Deserialize, Display, Eq, Hash, Ord, PartialEq, PartialOrd, Serialize)]
+pub enum MLSAGError {
```

Comment:
We need to create this error because otherwise we have a circular dependency:
* MLSAG Error (previously RingSignatureError) can be returned to `RingSigner` so `SignerError` has to possibly contain an MLSAG Error
* `RctBulletProofs::sign` calls the `RingSigner` now, so `SignerError` has to be convertible to `RingSignatureError`

When we split `MLSAG` into its own crate then this will be the error for that crate I think

Reply:
I like the way this is going where the proofing math isn't coupled to the core mobilecoin implementation details and we're having things like "RingSigner" which could be (in theory) be replaced with another valid underlying cryptosystem. I think as we're running into more blockchain integrations we want to support, it's almost necessary to decouple the systems to support the variety of interfaces out there.

Reply:
This should be `MlsagError`: https://rust-lang.github.io/api-guidelines/naming.html

Reply:
This is a rehash of comments here: https://github.com/mobilecoinfoundation/mobilecoin/pull/1977#discussion_r874103219
let's move any subsequent discussion of the name there